### DISCUSSION TITLE
WiFi networks treated as metered when using VPN

### DISCUSSION THREAD

**Discussion by mingyee2** [futo-org/grayjay-android#2290] [Ideas]
All WiFi networks I have tested in are treated as metered, for the purposes of determining both whether to download and the preferred video quality (resolution). Videos do download if the “WiFi & ethernet” option is chosen, unsurprisingly, but not when “Unmetered” is selected, where the video queued to be downloaded says “Waiting for unmetered” on the Downloads tab. All videos use the resolution chosen for “Preferred metered quality” both on data and WiFi networks.

All of the above persist even on WiFi networks that have the “Treat as unmetered” option selected in Android settings.

**Comment by mingyee2** [comment]
Update: still bugged on most recent patch

**Comment by mingyee2** [comment]
After some investigation, it appears this happens only when connected through a VPN. Still, it is behaviour that I wouldn’t expect based on the setting description, so it is probably still a bug and not a feature.

**Comment by Zvonimir-FUTO** [comment]
@mingyee2 Could you retest it with the latest version of Grayjay? I just tried to retest it with WireGuard VPN and couldn't reproduce the issue.

**Comment by mingyee2** [comment]
Still observing the same behaviour. I use Cloudflareʼs 1.1.1.1 VPN.

**Comment by Zvonimir-FUTO** [comment]
> Still observing the same behaviour. I use Cloudflareʼs 1.1.1.1 VPN.

Retested with Cloudflare's WARP and I can reproduce the issue. We will look into this.

**Comment by koen-futo** [comment]
On Android, VPN connections are typically treated as "metered" by default, meaning apps are more cautious about using data, especially in the background. This is because VPNs can use the mobile data connection if Wi-Fi is unavailable, potentially consuming your data allowance. While you can't directly change the "metered" status of a VPN, some workarounds may be possible depending on the specific VPN app and Android version.

You can always change the setting to allow metered connections to do more in the Grayjay app.
